contract_comm/caller.go

```diff
@@ -0,0 +1,119 @@
+	evm, err := c.builder.createEVM()
+	if err != nil {
+		return 0, err
+	}
+	return evm.CallFromSystem(contractAddress, abi, funcName, args, returnObj, gas, value)
+}
+
+func (c contractCommunicator) GetRegisteredAddress(registryId common.Hash) (*common.Address, error) {
+	evm, err := c.builder.createEVM()
+	if err != nil {
+		return nil, err
+	}
+	return vm.GetRegisteredAddressWithEvm(registryId, evm)
+}
+
+func (c contractCommunicator) MakeStaticCall(registryId common.Hash, abi abi.ABI, funcName string, args []interface{}, returnObj interface{}, gas uint64) (uint64, error) {
+	scAddress, err := c.getRegisteredAddress(registryId, funcName)
+	if err != nil {
+		return 0, err
+	}
+	// Record a metrics data point about execution time.
+	timer := metrics.GetOrRegisterTimer("contract_comm/systemcall/"+funcName, nil)
+	start := time.Now()
+	defer timer.UpdateSince(start)
+
+	gasLeft, err := c.MakeStaticCallWithAddress(scAddress, abi, funcName, args, returnObj, gas)
+	if err != nil {
+		log.Error("Error in executing function on registered contract", "function", funcName, "registryId", hexutil.Encode(registryId[:]), "err", err)
+	}
+	return gasLeft, err
+
+}
+
+func (c contractCommunicator) MakeCall(registryId common.Hash, abi abi.ABI, funcName string, args []interface{}, returnObj interface{}, gas uint64, value *big.Int) (uint64, error) {
+	scAddress, err := c.getRegisteredAddress(registryId, funcName)
+	if err != nil {
+		return 0, err
+	}
+	// Record a metrics data point about execution time.
+	timer := metrics.GetOrRegisterTimer("contract_comm/systemcall/"+funcName, nil)
+	start := time.Now()
+	defer timer.UpdateSince(start)
+
+	gasLeft, err := c.MakeCallWithAddress(scAddress, abi, funcName, args, returnObj, gas, value)
+	if err != nil {
+		log.Error("Error in executing function on registered contract", "function", funcName, "registryId", hexutil.Encode(registryId[:]), "err", err)
+	}
+	return gasLeft, err
+}
+
+// getRegisteredAddress gets the address from the registry and logs if the lookup fails
+func (c contractCommunicator) getRegisteredAddress(registryId common.Hash, funcName string) (common.Address, error) {
+	scAddress, err := c.GetRegisteredAddress(registryId)
+
+	if err != nil {
+		if err == ccerrors.ErrSmartContractNotDeployed {
+			log.Debug("Contract not yet registered", "function", funcName, "registryId", hexutil.Encode(registryId[:]))
+			return common.Address{}, err
+		} else if err == ccerrors.ErrRegistryContractNotDeployed {
+			log.Debug("Registry contract not yet deployed", "function", funcName, "registryId", hexutil.Encode(registryId[:]))
+			return common.Address{}, err
+		} else {
+			log.Error("Error in getting registered address", "function", funcName, "registryId", hexutil.Encode(registryId[:]), "err", err)
+			return common.Address{}, err
+		}
```

Comment:
Seems like we should have lifted the return statement out of the conditionals before.
```suggestion
		if err == ccerrors.ErrSmartContractNotDeployed {
			log.Debug("Contract not yet registered", "function", funcName, "registryId", hexutil.Encode(registryId[:]))
		} else if err == ccerrors.ErrRegistryContractNotDeployed {
			log.Debug("Registry contract not yet deployed", "function", funcName, "registryId", hexutil.Encode(registryId[:]))
		} else {
			log.Error("Error in getting registered address", "function", funcName, "registryId", hexutil.Encode(registryId[:]), "err", err)
		}
		return common.Address{}, err
```